src/redprl/judgment.fun

```diff
@@ -1,6 +1,6 @@
 functor SequentJudgment
   (structure S : SEQUENT where type 'a CJ.Tm.O.Ar.Vl.Sp.t = 'a list and type CJ.Tm.Sym.t = Sym.t and type CJ.Tm.Var.t = Sym.t
-   structure TermPrinter : SHOW where type t = S.CJ.Tm.abt) : LCF_JUDGMENT  =
+   structure TermPrinter : sig type t = S.CJ.Tm.abt val ppTerm : t -> Fpp.doc end) : LCF_JUDGMENT  =
```

Comment:
I think these sharing constraints are gnarly enough that we ought to instead consider "defunctorizing" this module if possible... I wonder if there were some circularities which made me do it this way?